percentage visitor wins and percentage home wins methods

andilovetto opened the pull request:
this has since been updated; we are no longer having the methods live within game_stats or game_factory. now the methods  are all living within stat_tracker and have been moved ~

instead of having the game_stats class hold the methods for game statistics, it seems better that the game_factory will hold the methods for the game statistics. the methods living in the game_stats class will just cause us to have to traverse farther down a filepath from stat_tracker and chain more things in order to call those methods.

andilovetto committed:
test: Add GameStats test spec file

andilovetto committed:
feat: Add GameStats file

andilovetto committed:
feat: Add GameStats file

andilovetto committed:
test: Remove test specs not needed

andilovetto committed:
Merge branch 'main' of github.com:ChrisAsercion/futbol into percentage_visitor_wins

andilovetto committed:
test: Add testing for percentage visitor wins and home wins

andilovetto committed:
test: Add requires for different lib files

andilovetto committed:
feat: Delete game stats file

andilovetto committed:
feat: Delete game stats spec file

andilovetto committed:
feat: Add variable assignment and new methods

andilovetto committed:
refactor: Remove additional spacing

ChrisAsercion reviewed (approved):
This looks good, moving forward I would say let's build methods in stat_tracker and then move them into their respective homes when refactoring later.

andilovetto committed:
Resolve merge conflicts - untrack coverage

andilovetto committed:
feat: Write best offense method and move existing methods to stat_tracker class. test: Write test for best offense method

andilovetto commented:
finished up writing and testing for the best offense method; moved all the methods that were written and living within game_factory into stat_tracker and changed the testing to reflect the same.

JorjaF commented on lib/game_factory.rb:
```diff
@@ -25,22 +22,4 @@ def create_games(path)
-      game[:away_goals] == game[:home_goals]
-
-    end
-    (ties.to_f / @games.length).round(2)
```
so this is where you deleted the percent of ties method on line 39-44

JorjaF commented on lib/stat_tracker.rb:
```diff
@@ -3,49 +3,81 @@
+      game[:away_goals] == game[:home_goals]
+    end
+  end
+
```
this is were you added that percent_of_ties method back in but some how you did not in the full method.

ChrisAsercion replied:
Is it just the one line  ` (ties.to_f / @games.length).round(2)` that this method needs? I don't see why it needs to be deleted out right and re-added with another branch. We can go in and add that line and then commit and this branch should work if I am understanding the problem correctly.

JorjaF reviewed (changes requested):
I found where in your deleted files where you took out my method and then in your add files you entered it back in but it was missing 3 lines. can you just go ahead and delete this method completely because I will just move it to stat_tracker

ChrisAsercion committed:
Adds line 51 back

JorjaF committed:
this is fined

JorjaF committed:
resolved conflics
